langserver/langserver_test.go

```diff
@@ -157,6 +157,7 @@ func TestServer(t *testing.T) {
 			fs: map[string]string{
 				"a.go":      "package p; var A int",
 				"a_test.go": `package p_test; import "test/pkg"; var X = p.A`,
```

Comment:
We should have a test case or another entry here that is `a2_test.go` `package p; var X = whatever` to make sure that packages `a` and `a_test` are kept separate.

If we have or add such a test case and it passes, then LGTM.